### PR TITLE
weather-forecast

### PR DESCRIPTION
**Pull request by Rayan84**
Dear reviewer:

This branch contains the weather forecast project, the application consists of: 

- Homepage: renders forecast of six well known cities, and filter options 
- Details page: renders the details of the weather forecast of the city selected by user

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,57 @@
-# Getting Started with Create React App
+# Weather Forecast
 
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+ This web application fetches weather forecast data from https://open-meteo.com/en/docs API for cities displayed on the homepage, the user can also choose specific country and then the city to view cooresponding temperature data.
 
-## Available Scripts
+ ## Live link
+ [Link](https://elastic-turing-eb476d.netlify.app/)
```
Comment (dasileker) [inline README.md:7]: - [x] Please record a video which will you tell about the project.

pppp

File: src/Redux/homepage/homepage.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+//       // do something with response
+//       users.push(response);
+//     })
+//   ));
+//   return users;
+// }
+
+// some other async context
+// console.log(await getMultiple({ id: 'asdf'}, { id: 'foo' }, { id: 'bar' }));
+
```
Comment (dasileker) [inline src/Redux/homepage/homepage.js]: - [x] please remove commented codes and unnecessary ones.

File: src/Redux/homepage/homepage.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  dispatch(fetchForecastRequest);
+  const cities = ['London', 'Amsterdam', 'Paris', 'New York', 'Tokyo', 'Cairo'];
+  const lat = [51.5072, 52.3676, 48.8566, 40.7128, 35.6762, 30.0444];
+  const lng = [0.1276, 4.9041, 2.3522, 74.0060, 139.6503, 31.2357];
+  const arr = [];
+
+  for (let i = 0; i < lat.length; i += 1) {
+    axios.get(`${FORECAST_URL}latitude=${lat[i]}&longitude=${lng[i]}&hourly=temperature_2m`, { headers: {} })
+      .then((response) => {
+      // arr.push(cities[i]);
```
Comment (dasileker) [inline src/Redux/homepage/homepage.js]: also this

File: src/Redux/homepage/homepage.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        const { data } = response;
+        const { hourly } = data;
+        /* eslint-disable-next-line */
```
Comment (dasileker) [inline src/Redux/homepage/homepage.js:49]: - [ ] please don't disable eslint, instead search on how to fix them

File: src/components/citiesIndexes.js [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+    cd: 'AF',
+    start: 24,
+    end: 334,
+  },
+  {
+    cd: 'AG',
+    start: 335,
+    end: 340,
+  },
+  {
```
Comment (dasileker) [inline src/components/citiesIndexes.js:22]: - [ ] instead of using an object to set cities in your select city, please use the props to make it.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Rayan84,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

- Your **app.test.js** is falling please fix the test and add more tests suits and add all the functions to your test like (data, city, hourly...)

Capture d’écran 2021-11-27 130413


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hi there, 

Awesome job on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet. Let's work on those.


 ## Required Changes ♻️
_Kindly check the comments under the review._


Happy coding, Cheers!🚀🚀

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it's valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Redux/details/details.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  type: FETCH_DETAILS_FAILURE,
+});
+
+export const fetchCityForecast = (lat, lng, cityName) => (dispatch) => {
+  dispatch(fetchForecastRequest());
+  axios.get(`${FORECAST_URL}latitude=${lat}&longitude=${lng}&hourly=temperature_2m`, { headers: {} })
+    .then((response) => {
+      const { data } = response;
+      const { hourly } = data;
+      /* eslint-disable-next-line */
```
Comment (mikenath223) [inline src/Redux/details/details.js:32]: - [ ] Please remove this eslint diable line its best to fix the issue in the code.

File: src/Redux/homepage/homepage.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  const lat = [51.5072, 52.3676, 48.8566, 40.7128, 35.6762, 30.0444];
+  const lng = [0.1276, 4.9041, 2.3522, 74.0060, 139.6503, 31.2357];
+  const arr = [];
+
+  for (let i = 0; i < lat.length; i += 1) {
+    axios.get(`${FORECAST_URL}latitude=${lat[i]}&longitude=${lng[i]}&hourly=temperature_2m`, { headers: {} })
+      .then((response) => {
+        const { data } = response;
+        const { hourly } = data;
+        /* eslint-disable-next-line */
```
Comment (mikenath223) [inline src/Redux/homepage/homepage.js:49]: - [x] Please remove this eslint diable line its best to fix the issue in the code.

File: src/components/TestComponent.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import returnObject from '../test/mock';
+import { test } from '../Redux/homepage/homepage';
+// import store from '../Redux/configureStore';
```
Comment (mikenath223) [inline src/components/TestComponent.js:5]: - [x] Kindly remove this commented code line.

File: src/index.js [hunk 9]
```diff
@@ -1,14 +1,20 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+import store from './Redux/configureStore';
+// import { loopFetchSixCities } from './Redux/homepage/homepage';
 
+// store.dispatch(loopFetchSixCities());
```
Comment (mikenath223) [inline src/index.js:10]: - [x] Please remove these commented code lines.

File: src/test/details.test.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+describe('Test making requests', () => {
+  it('Test make fetch request', () => {
+    expect(fetchForecastRequest()).toEqual({ type: 'detailsStore/details/fetch_request' });
+  });
+  it('Test fetch success', () => {
+    expect(fetchForecastSuccess('data')).toEqual({ type: 'detailsStore/details/fetch_success', payload: 'data' });
+  });
+  it('Test fetch failure', () => {
+    expect(fetchForecastFailure()).toEqual({ type: 'detailsStore/details/fetch_failure' });
```
Comment (mikenath223) [inline src/test/details.test.js:15]: - [x] Tests are still failing kindly fix them. Also please import actual components and test them, not a test component.

File: README.md [hunk 11]
```diff
@@ -1,70 +1,60 @@
...
 
-In the project directory, you can run:
+ ## Screenshot
+<img src='./src/laptop-screenshot.png'>
+<img src='./src/mobile-screenshot.png'>
 
-### `yarn start`
+## Presentation video
+[video](https://youtu.be/nw5vNCaCKn8)
```
Comment (mikenath223) [inline README.md:13]: - [x] In the video you are required to speak about a piece of code that you find interesting. Kindly record an additional video talking about it.

Comment (Rayan84) [reply README.md:13]: Thanks for the feedback, I think the presentation video isn't meant to get technical, it has to be very clear to people with no coding background.

**Pull request merged**